iXBRLViewerPlugin/iXBRLViewer.py

```diff
@@ -297,10 +297,16 @@ def addViewerToXMLDocument(self, xmlDocument, scriptUrl):
 
         for child in xmlDocument.getroot():
             if child.tag == '{http://www.w3.org/1999/xhtml}body':
+                alreadyWritten = False
                 for body_child in child:
-                    if body_child.tag == '{http://www.w3.org/1999/xhtml}script' and body_child.get('type','') == 'application/x.ixbrl-viewer+json':
-                        self.dts.error("viewer:error", "File already contains iXBRL viewer")
-                        return False
+                    if body_child.tag == '{http://www.w3.org/1999/xhtml}script':
+                        if body_child.get('type','') == 'application/x.ixbrl-viewer+json':
+                            alreadyWritten = True
+                        elif body_child.get('type', '') == 'text/javascript' and 'ixbrlviewer.js' in body_child.get('src'):
+                            body_child.set('src', scriptUrl)
+                if alreadyWritten:
+                    return True
```

Comment:
Out of curiosity, why this change?

Reply:
### tl;dr 
1. It is impossible to actually save the viewer with the current flow
2. the save viewer lets you change which version of the js file

### long winded explaination
If you have the plugin loaded in arelle and you are using the GUI the ixbrlViewer will automatically generate and open itself once you load a model [1](https://github.com/Workiva/ixbrl-viewer/blob/7bcde691a5edd59487e9e3158554dffb14c7886c/iXBRLViewerPlugin/__init__.py#L152), [2](https://github.com/Workiva/ixbrl-viewer/blob/7bcde691a5edd59487e9e3158554dffb14c7886c/iXBRLViewerPlugin/__init__.py#L136-L139). The viewer that it generates is using whatever `ixbrlviewer.js` is packaged with the plugin so in my case I am using the one I locally built `iXBRLVIewerPlugin/viewer/dist/ixbrlviewer.js`.  When you hit the save ixbrl viewer option in the arelle gui at this point (see the issue for screenshots) it will ask you for a link to the `ixbrlviewer.js` that you want to save using (at this point I switch to the [CDN server version](https://github.com/Workiva/ixbrl-viewer#accessing-via-the-cdn)). The viewer is only parsing the xmlDocument once and then tries to modify the same model (maybe a mistake?) when you try to hit save, but because it already generated a viewer prior to you hitting save (on document load) then it is currently stuck always returning `False` [here](https://github.com/Workiva/ixbrl-viewer/blob/7bcde691a5edd59487e9e3158554dffb14c7886c/iXBRLViewerPlugin/iXBRLViewer.py#L303) which then makes it skip saving [1](https://github.com/Workiva/ixbrl-viewer/blob/7bcde691a5edd59487e9e3158554dffb14c7886c/iXBRLViewerPlugin/iXBRLViewer.py#L368-L369), [2](https://github.com/Workiva/ixbrl-viewer/blob/7bcde691a5edd59487e9e3158554dffb14c7886c/iXBRLViewerPlugin/__init__.py#L100-L102)

Reply:
It seems wrong to reopen the already-ixbrl-viewer-enhanced XHTML file rather than reopening the source XHTML file when adding a different ixbrlviewer.js.  It seems like we should fix that rather than changing this error behavior.